lxd/device/disk.go

```diff
@@ -1035,3 +1047,63 @@ func (d *disk) getParentBlocks(path string) ([]string, error) {
 
 	return devices, nil
 }
+
+// generateVMConfigDrive generates an ISO containing the cloud init config for a VM.
+// Returns the path to the ISO.
+func (d *disk) generateVMConfigDrive() (string, error) {
+	scratchDir := filepath.Join(d.instance.DevicesPath(), strings.Replace(d.name, "/", "-", -1))
```

Comment:
Don't we already have some shared logic to figure out the device path?
That Replace above has the issue that someone who has `blah/abc` and a `blah-abc` devices will get a conflict.